### PR TITLE
Update to nuget 5.8.0 as earlier versions have compatibility bugs with VS 16.8 and later

### PR DESCRIPTION
**Pull request by dannyvv**
This PR updates the nuget version used to build this repo during CI, as well as the nuget version customers use when they build their projects via the react-native-cli.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6501)

### CODE REVIEW

File: packages/nuget-exe/package.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
   "name": "nuget-exe",
-  "version": "4.9.2",
+  "version": "5.8.0",
```
Comment (NickGerleman) [inline packages/nuget-exe/package.json:3]: Beachball will bump this to 5.8.1 as is. Changing this to `5.8.0-0` should create the right version though.

**Review by NickGerleman — APPROVED** [review]

**Review by asklar — APPROVED** [review]

**Comment by NickGerleman** [comment]
@dannyvv the E2ETest CI issue should be fixed if you merge master

**Comment by dannyvv** [comment]
Note from conversation with the MsBuild experts it seems msbuild now support restoring nuget packages from packages.config also during the build. I filed issue #6505 to investigate this.

**Comment by asklar** [comment]
@dannyvv  can we port this to 0.63?

**Comment by Khalef1** [comment]
Updating backport label. This was backported to 63 as #6507

**Pull request merged**